spec/tapioca/compilers/symbol_table_compiler_spec.rb

```diff
@@ -636,7 +636,9 @@ def foo
           M3 = Module.new
         RUBY
       ).to(
-        eq(<<~RUBY.chomp)
+        # this is a low impact hack to remove differences in Struct behaviour from
+        # ruby 2.4 testing.
+        eq(<<~RUBY.chomp.lines.reject { |line| RUBY_VERSION =~ /^2.4/ && line =~ /self.inspect/ }.join)
```

Comment:
Agreed, this is the only bit that feels weird. How about using ERB templates for this kind of output matching?

We can do something like:

```ruby
eq(render(<<~RUBY)
  class S1 < ::Struct
  def self.[](*_); end
<% unless Gem::Version.new(RUBY_VERSION) < Gem::Version.new("2.5.0") %>
  def self.inspect; end
<% end %>
  def self.members; end
  def self.new(*_); end
end
```

and `render` could be a method that takes a string and evaluates it as an ERB template.

Thoughts @Morriar?

Reply:
I definitely prefer this: https://github.com/Shopify/tapioca/pull/18/commits/f6b1129ebd19d4bb67bf3bf595e153ec85289c51

Reply:
The reason I propose this, btw, is because we have a similar situation in another test where the output changes depending on if you have Ruby compiled with GMP or not. 🤦‍♂

Reply:
@Morriar I saw that commit and that also looks good, but again, it only solves it for the ruby version problem and it ends up duplicating the test output. If we need to change one of the tests the change has to be duplicated etc, which is brittle.

Reply:
> ends up duplicating the test output

IMHO it will be easier to read / maintain than a test output riddled with ERB conditions.